### PR TITLE
fix(rule) : parentheses supported in expression

### PR DESCRIPTION
**Pull request by wKoza**
Issue #412 - part I

The second part of this issue concerns HTML entities. It's less trivial.

### CODE REVIEW

**Comment by mgechev** [comment]
There are some merge conflicts. Would you resolve them and merge afterwards?

File: src/angularWhitespaceRule.ts [hunk 1]
```diff
@@ -162,8 +162,16 @@ class PipeWhitespaceVisitor extends RecursiveAngularExpressionVisitor implements
 
 
     const replacements = [];
+    let parentheses = false;
+    let leftBeginning: number;
+    if(sf[exprEnd] === ')') {
```
Comment (mgechev) [inline src/angularWhitespaceRule.ts]: Can we have the parenthesis surrounded by whitespace?

**Comment by mgechev** [comment]
@wKoza left a single comment.

File: src/angularWhitespaceRule.ts (hunk 1 above)
Comment (wKoza) [reply src/angularWhitespaceRule.ts]: yes, we can. The rule doesn't check that.

**Comment by mgechev** [comment]
@wKoza travis complains because of linting errors:

```
ERROR: src/angularWhitespaceRule.ts[167, 7]: missing whitespace
ERROR: src/angularWhitespaceRule.ts[197, 9]: missing whitespace
ERROR: test/angularWhitespaceRule.spec.ts[233, 1]: trailing whitespace
ERROR: test/angularWhitespaceRule.spec.ts[249, 1]: trailing whitespace
```

**Pull request merged**